base-lite/ub/ub.go

```diff
@@ -0,0 +1,313 @@
+					if strings.HasPrefix(envKey, prefix) {
+						var effectiveKey string
+						if keep {
+							effectiveKey = envKey
+						} else {
+							effectiveKey = envKey[len(prefix)+1:]
+						}
+						config[ConvertKey(effectiveKey)] = envValue
+					}
+				}
+			}
+		}
+	}
+	return config
+}
+
+func writeConfig(writer io.Writer, config map[string]string) bool {
+	// Go randomizes iterations over map by design. We sort properties by name to ease debugging:
+	sortedNames := make([]string, 0, len(config))
+	for name := range config {
+		sortedNames = append(sortedNames, name)
+	}
+	sort.Strings(sortedNames)
+	for _, n := range sortedNames {
+		_, err := fmt.Fprintf(writer, "%s=%s\n", n, config[n])
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error printing configs: %s", err)
+			return false
+		}
+	}
+	return true
+}
+
+func loadConfigSpec(path string) (ConfigSpec, bool) {
+	var spec ConfigSpec
+	jsonFile, err := os.Open(path)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error opening json file %s : %s", path, err)
+		return spec, false
+	}
+	bytes, err := io.ReadAll(jsonFile)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error reading from json file %s : %s", path, err)
+		return spec, false
+	}
+
+	errParse := json.Unmarshal(bytes, &spec)
+	if errParse != nil {
+		fmt.Fprintf(os.Stderr, "Error parsing json file %s : %s", path, errParse)
+	}
+	return spec, true
+}
+
+func invokeJavaCommand(className string, jvmOpts string, args []string) bool {
+	classPath := getEnvOrDefault("UB_CLASSPATH", "/usr/share/java/cp-base-lite/*")
+
+	opts := []string{}
+	if jvmOpts != "" {
+		opts = append(opts, jvmOpts)
+	}
+	opts = append(opts, "-cp", classPath, className)
+	cmd := exec.Command("java", append(opts[:], args...)...)
+
+	if err := cmd.Run(); err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
+			return exitError.ExitCode() == 0
+		}
+		return false
+	}
+	return true
+}
+
+func getEnvOrDefault(envVar string, defaultValue string) string {
+	v, found := os.LookupEnv(envVar)
+	if !found {
+		return defaultValue
+	}
+	return v
+}
+
+func checkKafkaReady(minNumBroker string, timeout string, bootstrapServers string, zookeeperConnect string, configFile string, security string) bool {
+
+	opts := []string{minNumBroker, timeout + "000"}
+	if bootstrapServers != "" {
+		opts = append(opts, "-b", bootstrapServers)
+	}
+	if zookeeperConnect != "" {
+		opts = append(opts, "-z", zookeeperConnect)
+	}
+	if configFile != "" {
+		opts = append(opts, "-c", configFile)
+	}
+	if security != "" {
+		opts = append(opts, "s", security)
+	}
+	jvmOpts := os.Getenv("KAFKA_OPTS")
+	return invokeJavaCommand("io.confluent.admin.utils.cli.KafkaReadyCommand", jvmOpts, opts)
+}
+
+func checkAndPrintUsage(numArguments int, message string) {
+	if len(os.Args) != numArguments {
+		fmt.Fprintf(os.Stderr, "Usage '%s %s %s' \n", os.Args[0], os.Args[1], message)
+		os.Exit(1)
+	}
+}
+
+func main() {
+	success := false
+	if len(os.Args) < 2 {
```

Comment:
I'd suggest using a proper CLI flags/subcommand library, it will take care of usage and flags much easier, ex: https://github.com/spf13/cobra